keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,541 @@
+
+<p align="center">
+  <img src="mn-story-6.png?raw=true" alt="multi-network story 6 namespace networks profiles"/>
+</p>
+
+#### Story #7
+As a “Power User” with admin privileges I wish to have the ability to modify my
+Pod network namespace without any restrictions. I am aware that by doing this I
+might break established contracts for the kubernetes features.
+
+#### Story #8
+As a Virtual Machine -based compute platform provider that I run on top of
+Kubernetes and Kubevirt I need the ability to add/remove Network Interfaces to
+existing VMs without re-creating the VM. This would not be applicable for the
+Primary PodNetwork.
+
+<p align="center">
+  <img src="mn-story-8.png?raw=true" alt="multi-network story 8 network interface hot-plug"/>
+</p>
+
+#### Story #9
+As an infrastructure administrator I wish to support IPv6 SLAAC in the
+Kubernetes Pods. SLAAC's nature is to dynamically assign IP addresses and change
+them over time. Those should be reflected in Pod status. This would not be
+applicable for the Primary PodNetwork.
+
+### Requirements
+
+Below requirements are divided into phases. Each phase will be in its own design
+section of this KEP that will be defined and implemented incrementally.
+
+#### Phase I (base API and reference in Pod)
+1. Multi-network should not change the behavior of existing clusters where no 
+multi-network constructs are configured/used.
+2. Multi-network needs to be represented explicitly as an API construct in
+Kubernetes as it fundamentally ties in with other APIs in the Kubernetes
+networking model, extending their functionality.
+   * We will introduce an PodNetwork API object that represents the existing
+     infrastructure’s networking.
+3. PodNetwork functions as a handle that decouples the implementation
+(i.e. infrastructure/network admins) from the consumer (cluster user). This
+abstraction follows the Kubernetes model of decoupling intent from implementation.
+4. PodNetwork shall not define any implementation specific parameters in its
+specification.
+5. PodNetwork shall provide cluster generic options only.
+6. PodNetwork can reference to implementation-specific parameters.
+7. PodNetwork is the object Application developer will reference in their
+workflows (e.g. Pod, Service).
+8. Cluster Default PodNetwork is the PodNetwork the cluster has been created
+with initially.
+9. Cluster Default PodNetwork cannot be removed, but can be replaced.
+10. Pods can reference one or more PodNetworks when trying to attach to them.
+    * This list is explicit and all the PodNetworks that has to be available in
+      the Pod has to be listed.
+11. When Pods does not specify any reference to PodNetwork it connects to Cluster
+Default PodNetwork (network the cluster has been created with).
+12. Pod shall be able to provide additional configuration on how it attaches to
+a PodNetwork.
+13. Every Pod connected to specific PodNetwork must have connectivity within
```

Comment:
Pod with the same PodNetwork may also have multi-tenant usage, this way not all pods have connectivity within that network across the Cluster.

Reply:
How the multi-tenancy is achieved? via NetworkPolicy? if so, this is still valid, since NP is an explicit desire to limit connectivity, but in general this requirement still stands, w/o any extra configuration Pods within PodNetwork shall have connectivity.

Reply:
We achieve multi-tenancy via VPC. Ok, i get it.